src/erlcloud_kms.erl

```diff
@@ -0,0 +1,1060 @@
+
+
+-spec generate_data_key/2 ::
+          (KeyId :: string(),
+           Options :: generate_data_key_opts()) ->
+          any().
+generate_data_key(KeyId, Options) ->
+    generate_data_key(KeyId, Options, default_config()).
+
+
+-spec generate_data_key/3 ::
+          (KeyId :: string(),
+           Options :: generate_data_key_opts(),
+           Config :: aws_config()) ->
+          any().
+generate_data_key(KeyId, Options, Config) ->
+    OptJson = dynamize_options(Options),
+    Json = [{<<"KeyId">>, KeyId}|OptJson],
+    request(Config, "TrentService.GenerateDataKey", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%%------------------------------------------------------------------------------
+%%% GenerateDataKeyWithoutPlaintext
+%%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% KMS API:
+%% [http://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKeyWithoutPlaintext.html]
+%%
+%% ===Example===
+%%
+%%------------------------------------------------------------------------------
+-spec generate_data_key_without_plaintext/1 ::
+          (KeyId :: string()) ->
+          any().
+generate_data_key_without_plaintext(KeyId) ->
+    generate_data_key_without_plaintext(KeyId, []).
+
+
+-spec generate_data_key_without_plaintext/2 ::
+          (KeyId :: string(),
+           Options :: generate_data_key_opts()) ->
+          any().
+generate_data_key_without_plaintext(KeyId, Options) ->
+    generate_data_key_without_plaintext(KeyId, Options, default_config()).
+
+
+-spec generate_data_key_without_plaintext/3 ::
+          (KeyId :: string(),
+           Options :: generate_data_key_opts(),
+           Config :: aws_config()) ->
+          any().
+generate_data_key_without_plaintext(KeyId, Options, Config) ->
+    OptJson = dynamize_options(Options),
+    Json = [{<<"KeyId">>, KeyId}|OptJson],
+    request(Config, "TrentService.GenerateDataKeyWithoutPlaintext", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% GenerateRandom
+%%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% KMS API:
+%% [http://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateRandom.html]
+%%
+%% ===Example===
+%%
+%%------------------------------------------------------------------------------
+-spec generate_random/1 ::
+          (NumberOfBytes :: non_neg_integer()) ->
+          any().
+generate_random(NumberOfBytes) ->
+    generate_random(NumberOfBytes, default_config()).
+
+
+-spec generate_random/2 ::
+          (NumberOfBytes :: non_neg_integer(),
+           Config :: aws_config()) ->
+          any().
+generate_random(NumberOfBytes, Config) ->
+    Json = [{<<"NumberOfBytes">>, NumberOfBytes}],
```

Comment:
I think we should let the service do it. The end result is the same, but the other option results in more code to maintain.